docs/_ext/import_projects.py

```diff
@@ -145,12 +145,20 @@ def import_projects(app: Sphinx, config: Config):
         local_docs_path = local_dir / project.docs_path
         dest_docs_dir = srcdir / project.dest
 
-        # use the latest git tag when building docs
         if repo:
             tags = sorted(repo.tags, key=lambda t: t.commit.committed_datetime)
-            tag = tags[-1]
-            update_html_context(project, str(tag), str(tag.commit)[:7], config)
-            repo.git.checkout(str(tag))
+            if not tags:
+                # If tags don't exist just use the current commit. This occurs
+                # when the git repo fetch depth is 0.
```

Comment:
Based on the [actions/checkout readme](https://github.com/actions/checkout#usage) the force depth is 1 by default, which gives the one commit. I think the underlying issue is that the tags are not fetched. 

```suggestion
                # If tags don't exist just use the current commit. This occurs
                # when the git repo is a shallow clone.
```

In any case, I think setting `commit=repo.head.commit` is the correct behavior for the development builds. For dev builds, it is more informative to see the actual commits that generated the docs.

Reply:
updated the comment